functions/lambda_authorizer/app.py

```diff
@@ -48,8 +52,42 @@ def resource_to_arn_path(resource: str) -> str:
     return re.sub(r"\{[^}]+\}", "*", resource)
 
 
+def get_oidc_config(issuer: str) -> dict:
+    """Fetch and cache OIDC configuration per issuer"""
+    current_time = time.time()
+
+    if (
+        issuer in _oidc_config_cache
+        and (current_time - _oidc_config_cache_time.get(issuer, 0)) < jwks_cache_ttl
+    ):
+        return _oidc_config_cache[issuer]
+
+    # Validate the issuer URL scheme
+    parsed_issuer = urlparse(issuer)
+    if parsed_issuer.scheme not in ("https", "http"):
+        raise ValueError(f"Invalid issuer URL scheme: {parsed_issuer.scheme}")
+
+    discovery_url = f"{issuer}/.well-known/openid-configuration"
+
+    try:
+        response = requests.get(discovery_url, timeout=5)
+        response.raise_for_status()
+        config = response.json()
+
+        _oidc_config_cache[issuer] = config
+        _oidc_config_cache_time[issuer] = current_time
+
+        logger.info(
+            f"Fetched OIDC config for {issuer}, JWKS URI: {config.get('jwks_uri')}"
+        )
+        return config
+    except requests.exceptions.RequestException as e:
+        logger.warning(f"Failed to fetch OIDC config for {issuer}: {e}")
```

Comment:
Minor suggestion to use extra parameters for structured logging here rather than f-string substitution.

https://docs.aws.amazon.com/powertools/python/latest/core/logger/#extra-parameter